meteor/lib/mediaObjects.ts

```diff
@@ -169,557 +169,599 @@ export interface ScanInfoForPackage {
+							) {
+								messages.push({
+									status: PieceStatusCode.SOURCE_BROKEN,
+									message: generateTranslation('{{sourceLayer}} has {{audioStreams}} audio streams', {
+										sourceLayer: sourceLayer.name,
+										audioStreams,
+									}),
+								})
+							}
+							if (timebase) {
+								// check for black/freeze frames
+								const addFrameWarning = (arr: Array<PackageInfo.Anomaly>, type: string) => {
+									if (arr.length === 1) {
+										const frames = Math.ceil((arr[0].duration * 1000) / timebase)
+										if (arr[0].start === 0) {
+											messages.push({
+												status: PieceStatusCode.SOURCE_HAS_ISSUES,
+												message: generateTranslation(
+													'Clip starts with {{frames}} {{type}} frame',
+													{
+														frames,
+														type,
+														count: frames,
+													}
+												),
+											})
+										} else if (
+											mediaObject.mediainfo &&
+											mediaObject.mediainfo.format &&
+											arr[0].end === Number(mediaObject.mediainfo.format.duration) &&
+											(sourceDuration === undefined ||
+												Math.round(arr[0].start) * 1000 < sourceDuration)
+										) {
+											const freezeStartsAt = Math.round(arr[0].start)
+											messages.push({
+												status: PieceStatusCode.SOURCE_HAS_ISSUES,
+												message: generateTranslation(
+													'This clip ends with {{type}} frames after {{count}} second',
+													{
+														frames,
+														type,
+														count: freezeStartsAt,
+													}
+												),
+											})
+										} else if (
+											sourceDuration === undefined ||
+											Math.round(arr[0].start) * 1000 < sourceDuration
+										) {
+											messages.push({
+												status: PieceStatusCode.SOURCE_HAS_ISSUES,
+												message: generateTranslation(
+													'{{frames}} {{type}} frame detected within the clip',
+													{
+														frames,
+														type,
+														count: frames,
+													}
+												),
+											})
+										}
+									} else if (arr.length > 0) {
+										const dur = arr
+											.filter(
+												(a) => sourceDuration === undefined || a.start * 1000 < sourceDuration
+											)
+											.map((b) => b.duration)
+											.reduce((a, b) => a + b, 0)
+										const frames = Math.ceil((dur * 1000) / timebase)
+										if (frames > 0) {
+											messages.push({
+												status: PieceStatusCode.SOURCE_HAS_ISSUES,
+												message: generateTranslation(
+													'{{frames}} {{type}} frame detected in clip',
+													{
+														frames,
+														type,
+														count: frames,
+													}
+												),
+											})
+										}
+									}
+								}
+								if (!piece.content.ignoreBlackFrames && mediaObject.mediainfo.blacks?.length) {
+									addFrameWarning(mediaObject.mediainfo.blacks, 'black') // TODO - translate
```

Comment:
To fix the translations you could wrap the calls to generateTranslation as `generateFrameWarningTranslation(type, message, args)` and move the type from the args into the string. But since we are moving away from mediaObjects slowly I'm ok with leaving it untranslated, maybe check that with Rannveig though?

Reply:
I don't think the i18n tooling will handle that, which will result in no translations being produced for them.
The same flaw exists above/below for expected-packages.

I have an idea on how to solve it. by passing an object of the 'complete' strings into this function instead

Reply:
Your updated solution looks good 👍 (and along the lines of what I was thinking of)